log4j-api-test/src/test/java/org/apache/logging/log4j/util/PropertiesUtilTest.java

```diff
@@ -208,6 +208,28 @@ void testBadPropertySource() {
         }
     }
 
+    @Test
+    @ResourceLock(value = Resources.SYSTEM_PROPERTIES, mode = ResourceAccessMode.READ)
+    @Issue("https://github.com/apache/logging-log4j2/issues/3252")
+    void testRecursivePropertySource() {
```

Comment:
Shall we also verify the status logger warning?

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/3263/commits/149fcaed6ddd4d4ab23254df27ab25a62008d3e4.